### PR TITLE
The guzzlehttp/guzzle composer dependency has been updated to version 7.3.0

### PR DESCRIPTION
**Pull request by andrewbess**
### Description (*)

This pull request provides updating the `guzzlehttp/guzzle` composer dependency to version 7.3.0 for EE repository.

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/595

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32869

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by eduard13** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@andrewbess Some tests don't have information. May be it's related to related PR https://github.com/magento/partners-magento2ee/pull/595

**Comment by eduard13** [comment]
@magento run Unit Tests

**Comment by xmav** [comment]
@magento run Unit Tests with extensions eduard13/adobe-stock-integration:guzzlehttp-library-update

**Comment by eduard13** [comment]
This PR depends on https://github.com/adobe/stock-api-libphp/pull/21.

**Comment by eduard13** [comment]
@magento run Unit Tests

File: composer.json [hunk 1]
```diff
@@ -95,6 +95,12 @@
         "squizlabs/php_codesniffer": "~3.6.0",
         "symfony/finder": "^5.2"
     },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/eduard13/magento2-functional-testing-framework"
```
Comment (Den4ik) [inline composer.json]: Is this legal?

Comment (eduard13) [reply composer.json]: Personally, don't see any problem of using a such approach for testing purposes, since the MFTF changes weren't merged yet.

**Comment by eduard13** [comment]
@magento run all tests

File: composer.json (hunk 1 above)
Comment (Den4ik) [reply composer.json]: It's ok if PR marked as WIP, but little bit confuse when I see `Progress: review`

**Comment by xmav** [comment]
@magento run all tests

**Comment by andrewbess** [comment]
@magento run all tests

**Review by Den4ik — CHANGES_REQUESTED** [review]
Please check static tests

**Comment by andrewbess** [comment]
@magento run all tests

**Review by eduard13 — APPROVED** [review]

**Review by Den4ik — APPROVED** [review]
✅ Looks good for me

**Pull request merged**